File: app/src/main/java/de/markhaehnel/rbtv/rocketbeanstv/events/ScheduleLoadEvent.java
package de.markhaehnel.rbtv.rocketbeanstv.events;

import java.util.List;

import de.markhaehnel.rbtv.rocketbeanstv.objects.schedule.ScheduleItem;
import de.markhaehnel.rbtv.rocketbeanstv.utils.Enums.EventStatus;
import lombok.Getter;

@Getter
public class ScheduleLoadEvent {
    private List<ScheduleItem> shows;
    private EventStatus status;

    public ScheduleLoadEvent(EventStatus status) {
        this.status = status;
    }

    public ScheduleLoadEvent(List<ScheduleItem> shows, EventStatus status) {
        this.shows = shows;
        this.status = status;
    }
}

File: app/src/main/java/de/markhaehnel/rbtv/rocketbeanstv/events/ScheduleLoadEvent.kt
package de.markhaehnel.rbtv.rocketbeanstv.events

import de.markhaehnel.rbtv.rocketbeanstv.objects.schedule.ScheduleItem
import de.markhaehnel.rbtv.rocketbeanstv.utils.Enums.EventStatus

class ScheduleLoadEvent {
    var shows: List<ScheduleItem>? = null
    var status: EventStatus? = null

    constructor(status: EventStatus) {
        this.status = status
    }

    constructor(shows: List<ScheduleItem>, status: EventStatus) {
        this.shows = shows
        this.status = status
    }
}
